docs/src/userguide/wrapping_CPlusPlus.rst

```diff
@@ -545,6 +538,75 @@ listed in the table above. The code for this standard exception handler can be
 found `here
 <https://github.com/cython/cython/blob/master/Cython/Utility/CppSupport.cpp>`__.
 
+You can reuse Cython's built-in logic with something like the following.
+Notice the ``convert_current_exception_to_python()`` function. ::
+
+    from cpython.object cimport PyObject
+
+    cdef extern from *:
+        """
+        #include <Python.h>
+        #include <stdexcept>
+        #include <ios>
+
+        void rethrow_current_exception() {
+            throw;
+        }
+
+        int convert_current_exception_to_python();
+
+        void cpp_handle_exception(PyObject *exc_type) {
+            if (PyErr_Occurred())
+                return;
+            try {
+                throw;
+            }  catch (const std::logic_error& exn) {
+                PyErr_SetString(exc_type, exn.what());
+            } catch (...) {
+                convert_current_exception_to_python();
+            }
+        }
+        """
+
+        cdef void rethrow_current_exception() except+
+
+        cdef void cpp_handle_exception(object)
+
+    cdef public int convert_current_exception_to_python() except -1:
+        rethrow_current_exception()  # the heavy lifting is done in the ``except+``
```

Comment:
I think this can be a `void` function. It doesn't need Cython error propagation since it's not meant to be called from Cython. I'd also change the name a little to make the distinction between C++ exceptions and Python exceptions clearer.
```suggestion
    cdef public void convert_current_cpp_exception_to_python() except *:
        rethrow_current_cpp_exception()  # declared as ``except+``, which translates the exception
```